src/adapter/src/coord/message_handler.rs

```diff
@@ -54,11 +53,7 @@ impl Coordinator {
     /// callsite so that we can correctly instrument the boxed future here _and_
     /// so that we can stitch up the OpenTelemetryContext when we're processing
     /// a `Message::Command` or other commands that pass around a context.
-    pub(crate) fn handle_message<'a>(
-        &'a mut self,
-        span: tracing::Span,
-        msg: Message,
-    ) -> LocalBoxFuture<'a, ()> {
+    pub(crate) async fn handle_message<'a>(&'a mut self, span: tracing::Span, msg: Message) -> () {
         async move {
```

Comment:
We probably don't need this nested async block anymore, and instead you can annotate the async function with `#[instrument]`?